### ISSUE TITLE
lint vignette code

### ISSUE THREAD

**Issue by jackwasey** [jimhester/lintr#138]
It would be helpful to lint vignette code. For rmarkdown vignettes, I tried `knitr::purl`, and then linted the resulting `.R` file, which seemed okay at first glance. I expect other vignette formats would need different treatment.

The declarations in the `'''r somename someval = something'''` blocks could also be linted, but would need different default rules since they can't flow over two lines.

**Comment by jimhester** [comment]
this is already supported for all vignette types knitr supports (at least using the development version)

**Comment by jackwasey** [comment]
Works like a charm after installing github version. Thanks.

**Comment by dchiu911** [comment]
Calling `lintr::lint()` also lints the vignette code in `inst/doc/vignette_file_name.R`. It results in some unnecessary lints being called on the code chunks:
```r
## ----setup, echo=FALSE---------------------------------------------------
knitr::opts_chunk$set(
	message = FALSE,
	warning = FALSE,
	collapse = TRUE,
	comment = "#>",
	fig.align = "center",
	fig.width = 6,
	fig.height = 4.5
)
```
For example, for the above setup chunk, there would be a lint on Line 1: `Commented code should be removed`, and a lint on Line 3: `Put spaces around all  infix operators`, among many others.

**Comment by wlandau-lilly** [comment]
A couple things I notice:

1. `lint_package()` skips my `Rmd` vignettes by default.
1. Plain Markdown blocks like the ones below trigger an "Error: Malformed file!"

<pre><code>```
# code
```
</code></pre>
<pre><code>```r
# code
```
</code></pre>
